modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/granulecollector/ReprojectingSubmosaicProducerFactory.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * GeoTools - The Open Source Java GIS Toolkit
+ * http://geotools.org
+ *
+ * (C) 2016, Open Source Geospatial Foundation (OSGeo)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation;
+ * version 2.1 of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ */
+
+package org.geotools.gce.imagemosaic.granulecollector;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.geotools.gce.imagemosaic.RasterLayerRequest;
+import org.geotools.gce.imagemosaic.RasterLayerResponse;
+import org.geotools.gce.imagemosaic.RasterManager;
+
+/**
+ * Creates a SubmosaicProducer that can handle reprojecting granules into the mosaic's target CRS
+ */
+public class ReprojectingSubmosaicProducerFactory implements SubmosaicProducerFactory {
+    @Override
+    public List<SubmosaicProducer> createCollectors(RasterLayerRequest request,
```

Comment:
Btw, when this is done, make sure to also add a last sortby on the crs field to avoid jumps across different CRSs in granules having the same values for the main sort fields (e.g., if sorting on time to have most recent on top, make sure features with the same date are not in mixed crs order, but packed instead)